lib/net.js

```diff
@@ -397,22 +397,13 @@ function afterShutdown(status) {
 // Provide a better error message when we call end() as a result
 // of the other side sending a FIN.  The standard 'write after end'
 // is overly vague, and makes it seem like the user's code is to blame.
-function writeAfterFIN(chunk, encoding, cb) {
-  if (typeof encoding === 'function') {
-    cb = encoding;
-    encoding = null;
-  }
-
+Socket.prototype._writeAfterEnd = function(cb) {
   // eslint-disable-next-line no-restricted-syntax
   const er = new Error('This socket has been ended by the other party');
   er.code = 'EPIPE';
-  process.nextTick(emitErrorNT, this, er);
-  if (typeof cb === 'function') {
-    defaultTriggerAsyncIdScope(this[async_id_symbol], process.nextTick, cb, er);
-  }
+  defaultTriggerAsyncIdScope(this[async_id_symbol], process.nextTick, cb, er);
+};
 
-  return false;
-}
```

Comment:
What does changing this in ‘net’ achieves?

Reply:
Simplifies.